### PR TITLE
Adding TCs OCS-626 & 627: Create a StorageClass and make it as a default

### PR DESCRIPTION
**Pull request by PrasadDesala**
Adding OCS test case:  Create a StorageClass and make it as a default.
This PR covers testing this scenario on both RBD and Cephfs

Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

**Comment by PrasadDesala** [comment]
```
=========================================================================================== PASSES ===========================================================================================
================================================================================== short test summary info ===================================================================================
PASSED tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py::TestCreateStorageClassandMakeItAsDefault::test_create_sc_and_make_it_as_a_default[CephBlockPool]
PASSED tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py::TestCreateStorageClassandMakeItAsDefault::test_create_sc_and_make_it_as_a_default[CephFileSystem]
================================================================================= 2 passed in 604.98 seconds =================================================================================

```

File: ocs_ci/ocs/ocp.py [hunk 1]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
...
+            ret = default_sc_objs[i]['metadata']['annotations'][
+                'storageclass.kubernetes.io/is-default-class'
+            ]
+            if ret == 'true':
+                count = count + 1
+                default_storage_class.append(
+                    default_sc_objs[i]['metadata']['name'])
+    if count == 1:
+        log.info("One default StorageClass found")
+        return default_storage_class[0]
```
Comment (RazTamir) [inline ocs_ci/ocs/ocp.py]: you should return a list

File: ocs_ci/ocs/ocp.py [hunk 2]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
         bool: True on success, False otherwise
     """
     return switch_to_project(defaults.ROOK_CLUSTER_NAMESPACE)
+
+
+def get_default_storage_class():
```
Comment (RazTamir) [inline ocs_ci/ocs/ocp.py]: please move to helpers

File: ocs_ci/ocs/ocp.py [hunk 3]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
...
+        log.info("One default StorageClass found")
+        return default_storage_class[0]
+    elif count > 1:
+        log.info("More than one default StorageClass found")
+        return default_storage_class
+    else:
+        log.info("No default StorageClass found")
+
+
+def change_default_storageclass(scname):
```
Comment (RazTamir) [inline ocs_ci/ocs/ocp.py]: please move to helpers

File: ocs_ci/ocs/ocp.py [hunk 4]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
...
+    """
+    Change the default StorageClass to the given SC name
+
+    Args:
+        scname (str): StorageClass name
+
+    Returns:
+        bool: True on success, False otherwise
```
Comment (RazTamir) [inline ocs_ci/ocs/ocp.py]: not accurate - you are not returning False anywhere but an exception will be raised.
Not a big deal for now

File: tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        log.info(f"Current Default StorageClass is:{tmp_default_sc}")
+        assert tmp_default_sc == sc_obj.name, (
+            "Failed to change default StorageClass"
+        )
+        log.info(
+            f"Successfully changed the default StorageClass to "
+            f"{sc_obj.name}"
+        )
+
+        # Create a PVC using the default StorageClass
```
Comment (RazTamir) [inline tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py]: from this point on, it makes this test tier2

File: ocs_ci/ocs/ocp.py [hunk 6]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
...
+
+def get_default_storage_class():
+    """
+    Get the default StorageClass(es)
+
+    Returns:
+        list: StorageClass(es) list
+
+    """
+    count = 0
```
Comment (ebenahar) [inline ocs_ci/ocs/ocp.py]: Let's return the list of default storageclasses and examine it from the test. 
if we are expecting 1 default SC, then we can assert it from outside:

```python
assert len(default_storage_classes) > 1, "More than 1 storage class exist"
```

File: ocs_ci/ocs/ocp.py [hunk 7]
```diff
@@ -401,3 +401,64 @@ def switch_to_default_rook_cluster_project():
...
+    default_storage_class = []
+    default_sc_obj = OCP(kind='StorageClass')
```
Comment (ebenahar) [inline ocs_ci/ocs/ocp.py]: you can change lines 414-434 to:

```python
        default_sc_obj = OCP(kind='StorageClass')
        storage_classes = default_sc_obj.get().get('items')
        storage_classes = [sc for sc in storage_classes if 'annotations' in sc.get('metadata')]
        default_storage_classes = [
            sc for sc in storage_classes if sc
            .get('metadata').get('annotations')
            .get('storageclass.kubernetes.io/is-default-class') == 'true'
        ]
        return default_storage_classes
```

File: ocs_ci/ocs/ocp.py (hunk 2 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/ocp.py]: Done

File: ocs_ci/ocs/ocp.py (hunk 6 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/ocp.py]: Done

File: ocs_ci/ocs/ocp.py (hunk 1 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/ocp.py]: Done

File: ocs_ci/ocs/ocp.py (hunk 3 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/ocp.py]: done

File: ocs_ci/ocs/ocp.py (hunk 4 above)
Comment (PrasadDesala) [reply ocs_ci/ocs/ocp.py]: done

File: tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py (hunk 5 above)
Comment (PrasadDesala) [reply tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py]: Done

**Comment by ebenahar** [comment]
@PrasadDesala please add Jenkins execution

**Comment by PrasadDesala** [comment]
> @PrasadDesala please add Jenkins execution

Jenkins job keeps failing with one or other issue. Will re-trigger it again

**Comment by PrasadDesala** [comment]
```
=========================================================================================== PASSES ===========================================================================================
================================================================================== short test summary info ===================================================================================
PASSED tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py::TestCreateStorageClassandMakeItAsDefault::test_create_sc_and_make_it_as_a_default[CephBlockPool]
PASSED tests/manage/storageclass/test_create_sc_and_make_it_as_a_default.py::TestCreateStorageClassandMakeItAsDefault::test_create_sc_and_make_it_as_a_default[CephFileSystem]
================================================================================= 2 passed in 666.06 seconds =================================================================================

```

**Comment by ebenahar** [comment]
Verified:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/541/console

**Pull request merged**